### PR TITLE
p0f rewritten

### PR DESCRIPTION
**Pull request by hissssst**
* It is currently written in python3, but I will definetely rewrite it in six
* It's not much, because I don't want to find out that something went wrong in the beginning after writing many lines of code
* There are some comments for the reviewer. They will be definetely rewritten in the later versions of this file

### CODE REVIEW

File: scapy/modules/p0f.py [hunk 1]
```diff
@@ -1,126 +1,98 @@
-# This file is part of Scapy
-# See http://www.secdev.org/projects/scapy for more information
-# Copyright (C) Philippe Biondi <phil@secdev.org>
-# This program is published under a GPLv2 license
-
-"""
-Clone of p0f passive OS fingerprinting
-"""
-
```
Comment (gpotter2) [inline scapy/modules/p0f.py]: Could you leave the header ? Feel free to append your name to it :-)

File: scapy/modules/p0f.py [hunk 2]
```diff
@@ -1,126 +1,98 @@
...
-
-p0f_kdb, p0fa_kdb, p0fr_kdb, p0fo_kdb = None, None, None, None
 
+        #try:
+        self.base = {}
+        self.parse_file(f)
+        #except Exception:
+        #    warning("Can't parse p0f database (new p0f version ?)")
+        #    self.base = None
```
Comment (gpotter2) [inline scapy/modules/p0f.py]: You can get rid of this

File: scapy/modules/p0f.py [hunk 3]
```diff
@@ -1,621 +1,466 @@
...
+        olayout = packet2olayout(pkt)
+        quirks  = packet2quirks(pkt)
+        if pkt['TCP'].flags.A:
+            direction = 'response'
+        else:
+            direction = 'request'
+        gen = p0fdb.tcp_correl(direction, p0f_out, olayout, quirks)
+        return max(list(gen), key=lambda x: sum(x[0]))[1]
+
+if __name__ == '__main__':
```
Comment (p-l-) [inline scapy/modules/p0f.py]: I think this should go in a separate `.uts` file rather than here.

**Comment by guedou** [comment]
I added a reference to this PR in #399 and I am closing it due to no activity.

**Pull request closed without merging**